src/mmw/js/src/modeling/models.js

```diff
@@ -221,6 +221,54 @@ var ScenariosCollection = Backbone.Collection.extend({
             }
             model.set('active', active);
         });
+    },
+
+    duplicateScenario: function(cid) {
+        var source = this.get(cid),
+            sourceMods = source.get('modifications').models,
+            newModel = new ScenarioModel({
+                is_current_conditions: false,
+                name: this.makeNewScenarioName('Copy of ' + source.get('name'))
+            });
+
+        newModel.set('modifications', new ModificationsCollection(sourceMods, {
+            model: ModificationModel
+        }));
+
+        this.add(newModel);
+        this.setActiveScenario(newModel.cid);
+    },
+
+    // Generate a unique scenario name based off the baseName.
+    // Assumes a basic structure of "baseName X", where X is
+    // iterated as new scenarios with the same baseName are created.
+    // The first duplicate will not have an interated X in the name.
+    // The second duplicate will be "baseName 1".
+    makeNewScenarioName: function(baseName) {
```

Comment:
I'm working on #242 now and moved the logic from `Project` to here (it proved hard to test in it's current configuration because of the events), so now there is only once instance of this function.